tests/test_shell_completion.py

```diff
@@ -158,11 +163,20 @@ def custom(ctx, param, incomplete):
 def test_autocompletion_deprecated():
     # old function takes args and not param, returns all values, can mix
     # strings and tuples
-    def custom(ctx, args, incomplete):
+    def custom(
+        ctx: Context,
+        args: t.List[str],
+        incomplete: str,
+    ) -> t.Iterable[t.Union[t.Tuple[str, str], str]]:
         assert isinstance(args, list)
-        return [("art", "x"), "bat", "cat"]
+        yield "art", "x"
+        yield "bat"
+        yield "cat"
 
     with pytest.deprecated_call():
+        if t.TYPE_CHECKING:
+            # Force type checking against base class' constructor signature
+            Parameter(autocompletion=custom)
```

Comment:
The base class isn't instantiated, this doesn't need to be tested (and we're not checking tests either). Non-matching signatures should be picked up by mypy without specific tests.